applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+ *	                  or K_FOREVER to wait as long as necessary.
+ *
+ * @return 0 if successful, error value
+ */
+int amod_data_tx_rx(struct amod_handle handle_tx, struct amod_handle handle_rx,
+		    struct aobj_block *block_tx, struct aobj_block *block_rx, k_timeout_t timeout);
+
+/**
+ * @brief Helper function to configure the thread information for the module
+ *        set-up parameters structure.
+ *
+ * @param parameters   Pointer to the module set-up parameters
+ * @param description  Pointer to the modules private description
+ * @param stack        Memory block for the threads stack
+ * @param stack_size   Size of the threads stack
+ * @param priority     Priority of the thread insatnce, one of #amod_id
+ * @param msg_rx       Number of concurrent input messages
+ * @param msg_tx       Number of concurrent output messages
+ * @param data_slab    Slab to take a data item from
+ * @param data_size    Size of the data item that will be taken
+ *
+ * @return 0 if successful, error value
+ */
+int amod_parameters_configure(struct amod_parameters *parameters,
+			      struct amod_description *description, k_thread_stack_t *stack,
+			      size_t stack_size, int priority, struct data_fifo *msg_rx,
+			      struct data_fifo *msg_tx, struct k_mem_slab *data_slab,
+			      size_t data_size);
+
+/**
+ * @brief Helper function to return the base and instance names for a given
+ *        module handle.
+ *
+ * @param handle         A handle to this module instance
+ * @param base_name      A pointer to the name of the module
+ * @param instance_name  A pointer to the name of the current module instance
+ *
+ * @return 0 if successful, error value
+ */
+int amod_names_get(struct amod_handle *handle, char *base_name, char *instance_name);
+
+/**
+ * @brief Helper function to return the state of a given module handle.
+ *
+ * @param handle  A handle to this module instance
+ *
+ * @return 0 if successful, error value
+ */
+int amod_state_get(struct amod_handle *handle);
+
+/**
+ * @brief Helper function to attach a raw audio data buffer to an audio block.
+ *
+ * @param block        Pointer to the audio data block to fill
+ * @param data_type    The type of data carried in the block
+ * @param data         Pointer to the raw or coded audio data buffer
+ * @param data_size    Size of the raw or coded audio data buffer
+ * @param format       The format od the data carried in the block
+ * @param reference_ts Reference timestamp to be used to synchronise the data
+ * @param block_rx_ts  Block received timestamp to be used to synchronise the data
+ * @param bad_frame    A flag to indicate there are errors within the data for this block
+ * @param last_flag    A flag to indicate this is the last block in the stream
+ * @param user_data    A pointer to a private area of user data or NULL
+ *
+ * @return 0 if successful, error value
+ */
+int amod_block_data_attach(struct aobj_block *block, enum aobj_type data_type, char *data,
```

Comment:
What does "attach a raw audio data buffer to an audio block" mean in this context?

Reply:
An audio block is a known structure describing the data it is carrying. The data buffer will be the PCM or coded data that the module will work with. This is passed as a pointer so as to reduce copy overhead and hence this buffer is connected to the block by copying the pointer into the *data element.